tests/RobotFramework/config_management/child_conf_mgmt_plugin.robot

```diff
@@ -0,0 +1,333 @@
+Prerequisite Cloud
+    GET Parent ID                                       #Get the Parent ID from the cloud
+    GET Parent name                                     #Get the Parent name from the cloud
+Child device bootstrapping
+    Startup child device                                #Setting up/Bootstrapping of a child device
+Get child credentials
+    GET Child ID                                        #Get the Child ID from the cloud
+    GET Child name                                      #Get the Child name from the cloud
+    Validate child Name                                 #This is to check the existence of the bug: https://github.com/thin-edge/thin-edge.io/issues/1569
+Snapshot from device
+    Request snapshot from child device                  #Using the cloud command: "Get snapshot from device"
+    Child device response on snapshot request           #Child device is sending 'executing' and 'successful' MQTT responses
+    No response from child device on snapshot request   #Tests the failing of request after timeout of 10s  
+Child device config update
+    Send configuration to device                        #Using the cloud command: "Send configuration to device"
+    Child device response on update request             #Child device is sending 'executing' and 'successful' MQTT responses
+    No response from child device on config update      #Tests the failing of request after timeout of 10s  
+
+
+
+*** Keywords ***
+Create Timestamp
+    ${timestamp}=    get current date    result_format=%H%M%S
+    Set Suite Variable    ${timestamp}
+Define Child Device name
+    ${CHILD}=   Set Variable    CD${timestamp}
+    Set Suite Variable    ${CHILD}
+Parent Connection
+    Open Connection     ${PARENT_IP}
+    Login               ${USERNAME}    ${PASSWORD}
+Child Connection
+    Open Connection     ${CHILD_IP}
+    Login               ${USERNAME}    ${PASSWORD}
+Set external MQTT bind address
+    ${rc}=    Execute Command    sudo tedge config set mqtt.external.bind_address ${PARENT_IP}    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0}
+Set external MQTT port
+    ${rc}=    Execute Command    sudo tedge config set mqtt.external.port 1883    return_stdout=False    return_rc=True
+    Should Be Equal    ${rc}    ${0}
+Check for child related content
+    @{dir_cont}    List Directories In Directory    /etc/tedge/operations/c8y
+    Should Be Empty   ${dir_cont}
+    @{dir_cont}    List Directories In Directory    /etc/tedge/c8y
+    Should Be Empty   ${dir_cont}
+    @{dir_cont}    List Directories In Directory    /var/tedge
+    Should Be Empty   ${dir_cont}
+Delete child related content
+    Execute Command    sudo rm -rf /etc/tedge/operations/c8y/CD*         #if folder exists, child device will be created
+    Execute Command    sudo rm c8y-configuration-plugin.toml
+    Execute Command    sudo rm -rf /etc/tedge/c8y/*                    #if folder exists, child device will be created
```

Comment:
```suggestion
    Execute Command    sudo rm -rf /etc/tedge/c8y/CD*                    #if folder exists, child device will be created
```

Otherwise the parent's `c8y-configuration-plugin.toml` will also be deleted.

Reply:
Done